src/nupic/algorithms/Connections.cpp

```diff
@@ -55,6 +55,20 @@ void Connections::initialize(CellIdx numCells,
   maxSegmentsPerCell_ = maxSegmentsPerCell;
   maxSynapsesPerSegment_ = maxSynapsesPerSegment;
   iteration_ = 0;
+  nextEventToken_ = 0;
+}
+
+UInt32 Connections::subscribe(ConnectionsEventHandler* handler)
+{
+  UInt32 token = nextEventToken_++;
+  eventHandlers_[token] = handler;
+  return token;
+}
+
+void Connections::unsubscribe(UInt32 token)
```

Comment:
Would it be easier if `subscribe` took a `char*` name argument that you also pass to `unsubscribe`? That way the client just has to use the same string for both calls rather than keep track of the token returned.